File: example/javascript-gjs/fcitx/get-current-im.js
#!/usr/bin/env gjs

const GLib = imports.gi.GLib;
const Gio = imports.gi.Gio;


let bus = Gio.bus_get_sync(Gio.BusType.SESSION, null)
let rtn = bus.call_sync(
        'org.fcitx.Fcitx', //bus_name
        '/inputmethod', //object_path
        'org.fcitx.Fcitx.InputMethod', //interface_name
        'GetCurrentIM', //method_name
        null, // parameters
        GLib.VariantType.new('(s)'), //reply_type
        Gio.DBusCallFlags.NONE, //flags
        -1, //timeout_msec
        null //cancellable
)

//print(rtn.print(true));
print(rtn.deep_unpack()[0]);
 

File: example/python-pygi/dbus/fcitx/get-current-im.py
#!/usr/bin/env python3

import gi
gi.require_version('Gtk', '3.0')

from gi.repository import Gio, GLib

bus = Gio.bus_get_sync(Gio.BusType.SESSION, None)

rtn = bus.call_sync(
	bus_name = 'org.fcitx.Fcitx', 
	object_path = '/inputmethod', 
	interface_name = 'org.fcitx.Fcitx.InputMethod', 
	method_name = 'GetCurrentIM', 
	parameters = None, 
	reply_type = GLib.VariantType('(s)'), 
	flags = Gio.DBusCallFlags.NONE, 
	timeout_msec = -1, 
	cancellable = None
)

print(rtn.unpack()[0])

	

